### DISCUSSION TITLE
Penalty for hero death

### DISCUSSION THREAD

**Discussion by rnyoakum** [raceintospace/raceintospace#610] [Ideas]
After Yuri Gagarin was the first man in space, the Soviets essentially grounded him from the space program because he was too important to risk. I was thinking, what if RIS incorporated this type of consideration by adding an additional prestige or morale penalty if a hero astronaut/cosmonaut is killed on a mission?

On one hand, this could be interesting choice to model. On the other hand, RIS doesn't really need _another_ possible penalty. Thoughts?

**Comment by peyre** [comment]
That sounds like a good idea!  It would make sense that the deaths of those considered heroes would be harder on the director.  The game is much more forgiving than it used to be back in the floppy version (extreme runs of luck notwithstanding).

**Comment by hweimer** [comment]
Good idea, maybe this could be also combined with some (cash?) advantages if you have a hero on active duty or actually flying missions. How is the hero flag supposed to work anyway? The only benefit I could find is a mood bonus if you have at least two heros in a crew, but the way the hero flag gets awarded is unclear to me, to say the least.

**Reply by peyre** [reply]
Another good idea!  So using heroes would be taking a gamble.  I wonder though how this would work in practice - apart from Mercury/Vostok, your crews will have more than one.  If you fly a manned docking with Apollo, you now have a crew with three heroes, so would their next flight net us the bonus x3?  Or would it be bonus +1/hero after the first, or simply a flat number per crew?  (I assume extra penalties would work in whatever way bonuses work, whatever we settled on.)

**Reply by rnyoakum** [reply]
The Hero flag ensures the hero has a medal in the Space Museum view, which is otherwise reserved for astronauts/cosmonauts who completed 4+ missions. It also provides the mood bonus for two heroes in a crew, but that part feels off to me. I'm guessing it's a bug, and the morale bonus is supposed to apply _any_ time there is a hero in a flight crew.

**Reply by peyre** [reply]
That sounds right to me.

**Reply by hweimer** [reply]
The mood bonus is also applied to the heroes themselves, which is also a bit weird (or maybe it is because they are finally in a crew where they are not surrounded by complete amateurs ...).

I also finally figured out how the hero flag gets awarded: it's for Manned Space Mission, Manned Orbital, Manned Lunar Pass, Manned Lunar Landing, and (Lunar) Spacewalks, although the spacewalks only set the flag only for the crew that actually did it.
https://github.com/raceintospace/raceintospace/blob/e789738eb76eb9c0acf870c020961076e83342da/src/game/prest.cpp#L385
https://github.com/raceintospace/raceintospace/blob/e789738eb76eb9c0acf870c020961076e83342da/src/game/mc.cpp#L521
(hero contains the return value of HeroCheck)

**Reply by rnyoakum** [reply]
Or they get the hero bonus themselves because it's fun to be & to be lauded as a hero. Either way, requiring _two_ heroes in the same flight crew feels weird, one of those things that's undocumented and only comes up by accident. To me, it seems like a bug.

**Comment by zombiedesigner** [comment]
If I recall, hero’s garner additional prestige for every mission they go on, and a higher penalty in death too.   Or at least it’s in my new works Liftoff! 2.0
